lib/repl.js

```diff
@@ -282,17 +282,24 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
+    var out;
     internalUtil.decorateErrorStack(e);
-    if (e instanceof SyntaxError && e.stack) {
+
+    if (!((e instanceof Error) && typeof e.stack === 'string')) {
+      out = e;
```

Comment:
Thanks @addaleax. The special snowflake that is the REPL. I added a length check for `stack`, so errors with empty stacks will print with `util.inspect()`.